### ISSUE TITLE
Running experiments systematically in the cluster

### ISSUE THREAD

**Issue by sfmig** [SainsburyWellcomeCentre/crabs-exploration#145]
There are a few things to polish before we are able to systematically run experiments in the cluster. 

Experiments = training a detector with different (hyper)parameters, for at least 3 dataset splits, and comparing them.

Some bits noted so far:
- [ ] write a bash script for training in the cluster.
    - [ ] fix OOM issues
    - [ ] agree on handling of environments
 - [ ] Save checkpoints every (n?) epochs (right now, just saving the final model).
 - [ ] Save loss vs epoch, to monitor training.
 - [ ] Use validation set during training. Compute loss and eval metrics for it.
 - [ ] Link slurm job ID - MLflow run - final model (right now it's tricky to link them all)
 - [ ] Verify that we can overfit the training dataset. Right now the loss seems a bit high after ~100 epochs.
